addrmgr/addrmanager_test.go

```diff
@@ -558,3 +678,328 @@ func TestCorruptPeersFile(t *testing.T) {
 		t.Fatalf("Corrupt peers file has not been removed: %s", peersFile)
 	}
 }
+
+// TestValidatePeerNa tests whether a remote address is considered reachable
+// from a local address.
+func TestValidatePeerNa(t *testing.T) {
+	const unroutableIpv4Address = "0.0.0.0"
+	const unroutableIpv6Address = "::1"
+	const routableIpv4Address = "12.1.2.3"
+	const routableIpv6Address = "2003::"
+	onionCatTorV2Address := onionCatNet.IP.String()
+	rfc4380IPAddress := rfc4380Net.IP.String()
+	rfc3964IPAddress := rfc3964Net.IP.String()
+	rfc6052IPAddress := rfc6052Net.IP.String()
+	rfc6145IPAddress := rfc6145Net.IP.String()
+
+	tests := []struct {
+		name          string
+		localAddress  string
+		remoteAddress string
+		valid         bool
+		reach         int
+	}{
+		{
+			name:          "torv2 to torv2",
+			localAddress:  onionCatTorV2Address,
+			remoteAddress: onionCatTorV2Address,
+			valid:         false,
+			reach:         Private,
+		},
+		{
+			name:          "routable ipv4 to torv2",
+			localAddress:  routableIpv4Address,
+			remoteAddress: onionCatTorV2Address,
+			valid:         true,
+			reach:         Ipv4,
+		},
+		{
+			name:          "not routable ipv4 to torv2",
```

Comment:
Perhaps "not routable" in the test names could be "unroutable".

Reply:
Right, that's more consistent with the rest of the code. Thanks, updated.